### PR TITLE
[FLINK-22451][table] Support (*) as parameter of UDFs in Table API

### PR DESCRIPTION
**Pull request by yittg**
## What is the purpose of the change

Support (*) as parameter of UDFs in Table API, it will be expanded as all columns of the table.

## Verifying this change

This change added tests and can be verified as follows:

  - An unittest case in ExpressionResolverTest.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (no)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (no)
  - The serializers: (no)
  - The runtime per-record code paths (performance sensitive): (no)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn/Mesos, ZooKeeper: (no)
  - The S3 file system connector: (no)

## Documentation

  - Does this pull request introduce a new feature? (yes)
  - If yes, how is the feature documented? (docs)

### CODE REVIEW

File: flink-table/flink-table-planner-blink/src/test/java/org/apache/flink/table/planner/functions/CallFunctionITCase.java [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.apache.flink.table.api.Expressions.$;
+import static org.apache.flink.table.api.Expressions.call;
+import static org.hamcrest.CoreMatchers.equalTo;
+import static org.junit.Assert.assertThat;
+
+/** IT tests for call functions. */
+public class CallFunctionITCase {
```
Comment (wuchong) [inline flink-table/flink-table-planner-blink/src/test/java/org/apache/flink/table/planner/functions/CallFunctionITCase.java:46]: Could you add this test into `org.apache.flink.table.planner.runtime.stream.table.CalcITCase`? We should avoid introducing test classes for each small feature to reduce the maintain burden.

Comment (yittg) [reply flink-table/flink-table-planner-blink/src/test/java/org/apache/flink/table/planner/functions/CallFunctionITCase.java]: @wuchong  Thanks, i have fixed it.

File: flink-table/flink-table-planner-blink/src/test/scala/org/apache/flink/table/planner/runtime/stream/table/CalcITCase.scala [hunk 2]
```diff
@@ -529,6 +529,27 @@ class CalcITCase(mode: StateBackendMode) extends StreamingWithStateTestBase(mode
...
+
+  /**
+   * This is an edition of [[CalcITCase.testMap()]] that uses * as one argument of the UDFs.
+   */
+  @Test
+  def testUserDefinedFunctionWithStarParameter(): Unit = {
+    val ds = env.fromCollection(smallTupleData3).toTable(tEnv, 'a, 'b, 'c)
+      .map(Func23('*)).as("a", "b", "c", "d")
+      .map(Func24('*)).as("a", "b", "c", "d")
+      .map(Func1('b))
```
Comment (wuchong) [inline flink-table/flink-table-planner-blink/src/test/scala/org/apache/flink/table/planner/runtime/stream/table/CalcITCase.scala:541]: Could you also test `where`, `select` too? I remember you tested it in the original test.

**Comment by yittg** [comment]
@wuchong Please have another look.

**Comment by yittg** [comment]
@wuchong Hi, how's everything going?

**Review by wuchong — APPROVED** [review]
LGTM.

I rebased your branch to trigger the build again. Will merge it once build is passed.

**Comment by yittg** [comment]
Great, really appreciate for you help. @wuchong

**Pull request merged**